src/shared/parse-helpers.c

```diff
@@ -56,11 +56,12 @@ int path_simplify_and_warn(
                                   "%s= path is not normalized%s: %s",
                                   lvalue, fatal ? "" : ", ignoring", path);
 
-        if (FLAGS_SET(flags, PATH_CHECK_NON_API_VFS) && path_below_api_vfs(path))
-                return log_syntax(unit, level, filename, line, SYNTHETIC_ERRNO(EINVAL),
-                                  "%s= path is below API VFS%s: %s",
-                                  lvalue, fatal ? ", refusing" : ", ignoring",
-                                  path);
+        if ((flags & (PATH_CHECK_NON_API_VFS | PATH_CHECK_NON_API_VFS_DEV_OK)) && path_below_api_vfs(path))
+                if (!((flags & PATH_CHECK_NON_API_VFS_DEV_OK) && path_startswith(path, "/dev")))
```

Comment:
Here FLAGS_SET can be used.

Reply:
i think it might make sense to to split out this check, i.e. add a tiny helper call:


```c
static bool validate_api_vfs(const char *path, PathSimplifyWarnFlags flags) {

        if ((flags & (PATH_CHECK_NON_API_VFS|PATH_CHECK_NON_API_VFS_DEV_OK)) == 0)
                return true;

        if (!path_below_api_vfs(path))
                return true;

        return !FLAGS_SET(PATH_CHECK_NON_API_VFS_DEV_OK) || path_startswith(path, "/dev"));
}
```

or so? i think that would make the thing a lot more readable.

Reply:
Thanks, added the helper with IMHO even more readable final check.